File: source/RolXServer/RolXServer/Projects/WebApi/Resource/SubprojectShallow.cs
// -----------------------------------------------------------------------
// <copyright file="SubprojectShallow.cs" company="Christian Ewald">
// Copyright (c) Christian Ewald. All rights reserved.
// Licensed under the MIT license.
// See LICENSE.md in the project root for full license information.
// </copyright>
// -----------------------------------------------------------------------

namespace RolXServer.Projects.WebApi.Resource;

public record SubprojectShallow(
    int Id,
    string FullNumber,
    string CustomerName,
    string ProjectName,
    string Name,
    string ManagerName,
    bool IsClosed);

File: source/rolx-client/src/app/projects/core/subproject-shallow.ts
import { assertDefined } from '@app/core/util/utils';

export class SubprojectShallow {
  id!: number;
  fullNumber!: string;
  customerName!: string;
  projectName!: number;
  name!: string;
  managerName!: string;
  isClosed!: boolean;

  validateModel(): void {
    assertDefined(this, 'id');
    assertDefined(this, 'fullNumber');
    assertDefined(this, 'customerName');
    assertDefined(this, 'projectName');
    assertDefined(this, 'name');
    assertDefined(this, 'managerName');
    assertDefined(this, 'isClosed');
  }
}
